src/lib/array.jl

```diff
@@ -245,14 +265,32 @@ for t in subtypes(AbstractWorkerPool)
 end
 @nograd workers
 
+collect_if_dict(x::Dict) = [x for x in x], collect(keys(x))
+collect_if_dict(x) = x, nothing
+
+reconstruct_if_dict(x̄, _keys::Nothing) = x̄
+
+function reconstruct_if_dict(x̄, _keys)
```

Comment:
Maybe deserves a comment, as the logic of dispatch on _keys wasn't obvious on first glance:
```suggestion
# This reverses `collect_if_dict`, which returns `_keys::Nothing` if x is not a Dict
reconstruct_if_dict(x̄, _keys::Nothing) = x̄
function reconstruct_if_dict(x̄, _keys)
```